packages/web/src/features/translation/TranslationSidebar.tsx

```diff
@@ -166,14 +166,13 @@ export const TranslationSidebar = ({
                   <RichTextInput
                     key={word.id}
                     name="noteContent"
-                    value={originalNoteContent}
+                    value={noteContent}
                     onBlur={async (e) => {
-                      setNoteContent(e.target.value);
-                      saveNote();
+                      saveNote(e.target.value);
+                      saveNote.flush();
                     }}
                     onChange={async (e) => {
-                      setNoteContent(e.target.value);
-                      debouncedSaveNote();
+                      saveNote(e.target.value);
                     }}
```

Comment:
Now we don't have to keep track of any other state, we just have the `noteContent` which changes when the selected word changes and resets the rich text input. But changes just get propagated to the server without having to be synced with react state. The rich text input owns the current state rather than the sidebar.